stellargraph/layer/node2vec.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2019 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+node2vec
+
+"""
+__all__ = ["Node2Vec"]
+
+from tensorflow.keras import Input
+from tensorflow.keras import backend as K
+from tensorflow.keras.layers import Reshape, Embedding
+import math
+from tensorflow import keras
+import warnings
+
+from ..mapper import Node2VecLinkGenerator, Node2VecNodeGenerator
+
+
+class Node2Vec:
+    """
+    Implementation of the node2vec algorithm of A. Grover and J. Leskovec with Keras layers.
+    see: https://snap.stanford.edu/node2vec/
+
+    The model minimally requires specification of the embedding size and a generator object.
+
+    Args:
+        emb_size (int): The dimension of node embeddings.
+        generator (Sequence): A NodeSequence or LinkSequence.
+        node_num (int): The number of nodes in the given graph.
+    """
+
+    def __init__(self, emb_size, generator=None, **kwargs):
+
+        # Get the node_num from the generator if it is given
+        self.generator = generator
```

Comment:
Do we need to check that the generator type is compatible?

Reply:
the check is done in the function of _get_sizes_from_generator